R/draws.R

```diff
@@ -1,10 +1,97 @@
-#' TODO
+#' @title Fit the base imputation model and get parameter estimates
+#'
+#' @description fits the base imputation model to the observed data assuming MAR
+#' according to the given multiple imputation methodology and returns the parameter
+#' estimates from the fit.
 #'
 #' @name draws
-#' @param data TODO
-#' @param data_ice TODO
-#' @param vars TODO
-#' @param method TODO
+#' @param data A `data.frame` containing the data to be used in the model. See details.
+#' @param data_ice A `data.frame` that specifies the information related
+#' to the intercurrent events (ICE) and the imputation strategies. See details.
+#' @param vars A `vars` object as generated by [set_vars()]. See details.
+#' @param method A `method` object as generated by either [method_bayes()],
+#' [method_approxbayes()] or [method_condmean()].
+#' It specifies the multiple imputation methodology to be used. See details.
+#'
+#' @details
+#'
+#' `draws` performs the first step of the multiple imputation (MI) procedure: fitting the
+#' base imputation model. The goal is to estimate the parameters of interest needed
+#' for the imputation phase (i.e. the regression coefficients and the covariance matrices
+#' from a MMRM model).
+#'
+#' The function distinguishes between the following methods:
+#' - Bayesian MI based on MCMC sampling: `draws` returns the draws
+#' from the posterior distribution of the parameters using a Bayesian approach based on
+#' MCMC sampling. This method can be specified by using `method = method_bayes()`.
+#' - Approximate Bayesian MI based on bootstrap: `draws` returns
+#' the draws from the posterior distribution of the parameters using a Bayesian approach,
```

Comment:
Is "using bayesian approach" potentially confusing here ?

Reply:
I can use "using an approximate Bayesian approach"?

Reply:
sounds good to me